internal/core/src/storage/PayloadReader.cpp

```diff
@@ -18,35 +18,67 @@
 #include "exceptions/EasyAssert.h"
 #include "storage/FieldDataFactory.h"
 #include "storage/Util.h"
+#include "log/Log.h"
+#include "parquet/column_reader.h"
+#include "arrow/io/api.h"
+#include "arrow/status.h"
+#include "parquet/arrow/reader.h"
 
 namespace milvus::storage {
-PayloadReader::PayloadReader(std::shared_ptr<PayloadInputStream> input, DataType data_type) : column_type_(data_type) {
-    init(input);
-}
 
 PayloadReader::PayloadReader(const uint8_t* data, int length, DataType data_type) : column_type_(data_type) {
-    auto input = std::make_shared<storage::PayloadInputStream>(data, length);
+    auto input = std::make_shared<arrow::io::BufferReader>(data, length);
     init(input);
 }
 
 void
-PayloadReader::init(std::shared_ptr<PayloadInputStream> input) {
-    auto mem_pool = arrow::default_memory_pool();
-    // TODO :: Stream read file data, avoid copying
-    std::unique_ptr<parquet::arrow::FileReader> reader;
-    auto st = parquet::arrow::OpenFile(input, mem_pool, &reader);
-    AssertInfo(st.ok(), "failed to get arrow file reader");
-    std::shared_ptr<arrow::Table> table;
-    st = reader->ReadTable(&table);
-    AssertInfo(st.ok(), "failed to get reader data to arrow table");
-    auto column = table->column(0);
-    AssertInfo(column != nullptr, "returned arrow column is null");
-    AssertInfo(column->chunks().size() == 1, "arrow chunk size in arrow column should be 1");
-    auto array = column->chunk(0);
-    AssertInfo(array != nullptr, "empty arrow array of PayloadReader");
-    dim_ = datatype_is_vector(column_type_) ? GetDimensionFromArrowArray(array, column_type_) : 1;
-    field_data_ = FieldDataFactory::GetInstance().CreateFieldData(column_type_, dim_);
-    field_data_->FillFieldData(array);
+PayloadReader::init(std::shared_ptr<arrow::io::BufferReader> input) {
+    arrow::MemoryPool* pool = arrow::default_memory_pool();
+
+    // Configure general Parquet reader settings
+    auto reader_properties = parquet::ReaderProperties(pool);
+    reader_properties.set_buffer_size(4096 * 4);
+    reader_properties.enable_buffered_stream();
+
+    // Configure Arrow-specific Parquet reader settings
+    auto arrow_reader_props = parquet::ArrowReaderProperties();
+    arrow_reader_props.set_batch_size(128 * 1024);  // default 64 * 1024
```

Comment:
any reason? 
for vector datas 128 *1024 seems to huge?
what about 16 * 1024

Reply:
batch data size  not  larger than file size 16m